ivy_tests/test_ivy/test_frontends/test_torch/test_creation_ops.py

```diff
@@ -150,6 +150,60 @@ def _start_stop_step(draw):
 # ------------ #
 
 
+# complex
+@handle_frontend_test(
+    fn_tree="torch.complex",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float")),
+)
```

Comment:
Now, about the issue with the `out` argument when adding `@to_ivy_arrays_and_back`. I believe there is some extra logic that needs to be added to our frontend design to handle the `out` argument properly, so there is no expectation to have that working in your PR. You can go ahead and add `test_with_out=st.just(False)` here to exclude `out` from the test.

Reply:
ok, I might go with that. 
Would it be a problem if I implemented `ivy.complex` on this PR? the logic itself is pretty simple, and I personally prefer doing that then "bandaging" the test itself.

Reply:
ok, I might go with that. 
Would it be a problem if I implemented `ivy.complex` on this PR? The logic itself is pretty simple, and I personally prefer doing that than "bandaging" the test itself.

Reply:
Looking into it a bit more, it seems that the reason is that using `@to_ivy_arrays_and_back` makes it only work with frontend arrays, not working with `ivy.array` either (when using `out`). It seems to be more of a structural problem, since using an already implemented method like `torch.as_array` also yields an error (example below):

```python
import ivy
import ivy.functional.frontends.torch as torch

a = ivy.array([1,2,3], dtype=torch.float64)
b = ivy.array([2,3,4], dtype=torch.float64)
c = torch.as_tensor(a, out=b)
```

The error doesn't happen when using `torch.tensor`. I guess it doesn't make a difference whether `ivy.complex` exists, so I'll go with the more straightforward approach for now. Would it be okay to open an issue for the weird `out` behavior tho?

Reply:
`torch.as_tensor` and `torch.tensor` don't have an `out` argument at all actually. I will make a note to move the `out` assertion so that it doesn't come up in such cases before the "argument not supported" error. It is misleading.

Reply:
The `out` argument of frontend functions is handled by the ivy function called inside them. 
Essentially `@to_ivy_arrays_and_back` on `torch.complex` will convert the `out` frontend array to an ivy array, that ivy array will get passed to `ivy.complex`, and finally the `@handle_out_argument` on `ivy.complex` will do the inplace update on the `out` array.

Reply:
Yeah, I've messed up with that example, my bad.

Dragging this a bit longer tho, the error still happens when using `ivy.array` for `out` in functions that do support `out` (tested it with `torch.full` and `torch.eye`, they both use `ivy` impls, support `out` and raise the same error), just felt it could be good to notify you guys.

Regardless, this seems to be out of the scope of this PR at this point. I've recently updated the PR adding the `@to_ivy_arrays_and_back` to `torch.polar`, so it seems to be good to go.